test/strategy/ExtendConfigStrategyTest.js

```diff
@@ -7,6 +7,43 @@ var strategy = require('../../lib/strategy');
 var ExtendConfigStrategy = strategy.ExtendConfigStrategy;
 
 describe('ExtendConfigStrategy', function () {
+  it("should be able to extend member that references itself", function (done) {
+    var extendConfig = {
+      abc: "123"
+    };
+
+    extendConfig.root = extendConfig;
+    extendConfig.something = {
+      root: extendConfig
+    };
+
+    var strategy = new ExtendConfigStrategy(extendConfig);
+
+    strategy.process({}, function(err, config) {
+      if (err) {
+        throw err;
+      }
+
+      assert.deepEqual(config, {
+        abc: "123",
```

Comment:
Consistent to the rest of the library;) Will change this when we have a .eslint in place.